### PR TITLE
Pull BTF information if missing using BTFHub

### PR DESCRIPTION
**Pull request by kakkoyun**
This PR enables the agent to run in a valid `CO:RE` fashion. By these changes, the eBPF programs will be able to run flawlessly for different Kernel versions and different architectures (even though we only support x86 right now, soon we will add ARM64 support).

To be able to achieve this goal:
- It downloads the supported Kernel BTF definitions from BTF Hub during builds time, embeds them to the binary, and loads those definitions in the case of missing BTF definitions for a kernel on a running host. The host's BTF definition is always prioritized.
- It converts all the helper functions with CO:RE equivalents.

Moreover, PR makes changes to the build pipeline to support building multiple eBPF objects files and consuming them to generate necessary BTF files using different definitions for corresponding Kernel versions.
 
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

Fixes https://github.com/parca-dev/parca-agent/issues/715

Related/Blocker for: #1209 #697 #1114

### TODO:
* [x] Add CI cache for "btfhub-archive" downloads
* [x] Cross-generate BTF for arm/x86
* [x] Support multiple eBPF object files
* [x] Add missing CI tooling: `bpftool`, ?, ...
* [x] Check if the CI host machines are BTF enabled
* [x] Modify the eBPF program to read from the task struct
  * [x] PT_REGS
  * [x] (interim or test only) task_struct->PID 
* [ ] Testing (using temporary task_struct->PID)
  * [x] Test on x86
  * [x] Test on CI
  * [ ] Test on arm64 (with lima or making CI work on ARM)
  * [ ] Check error cases when BTF files are missing

### CODE REVIEW

File: bpf/profiler/cpu.bpf.c [hunk 1]
```diff
@@ -670,9 +670,9 @@ static __always_inline void set_initial_state(bpf_user_pt_regs_t *regs) {
   // we aren't reading garbage data.
   unwind_state->stack.len = 0;
 
-  unwind_state->ip = regs->ip;
-  unwind_state->sp = regs->sp;
-  unwind_state->bp = regs->bp;
+  unwind_state->ip = PT_REGS_IP_CORE(regs);
+  unwind_state->sp = PT_REGS_SP_CORE(regs);
+  unwind_state->bp = PT_REGS_FP_CORE(regs);
```
Comment (javierhonduco) [inline bpf/profiler/cpu.bpf.c:675]: Niiiiiiiice!! ❤️

**Comment by kakkoyun** [comment]
I'll pick this up after merging #1142. We need to add tests.

**Comment by kakkoyun** [comment]
After the changes in #1805, this will be easier to test.
I will start a new PR soon rather than lose my hair while rebasing this branch.

**Comment by kakkoyun** [comment]
I don't think it worths resolving the conflicts.

**Pull request closed without merging**